docs/dev/building.md

```diff
@@ -22,37 +22,44 @@ git clone https://github.com/scylladb/scylla
 cd ./scylla
 git submodule update --init --force --recursive
 ./tools/toolchain/dbuild ./configure.py --mode=<mode>
-./tools/toolchain/dbuild ninja
+./tools/toolchain/dbuild cmake --build build
```

Comment:
Wait, what? Aren't you supposed to continue running "ninja" to build? Also in README.md you gave an example building with ninja, so why is this one different?
I suggest that we:
1. Think of "the best" way to build - if you think it's with cmake, not ninja, than give this example in all doc files and not just one.
2. In one (or at least one) doc files also give the "other" way, in case someone is curious.

I see other reviewers commented this too. I think they were right.

Reply:
i think it's cmake. but per your comment at https://github.com/scylladb/scylladb/pull/20188/files#r1913248685 , it seems we need to stick with `ninja`. so i am using `ninja` in all doc, and provide `cmake` in `building.md` for those who are curious.